File: Contests/Biweekly/Biweekly Contest 97/2554. Maximum Number of Integers to Choose From a Range I.java
/*
Problem Link: https://leetcode.com/contest/biweekly-contest-97/problems/maximum-number-of-integers-to-choose-from-a-range-i/
*/

import java.util.*;

class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < banned.length; i++) {
            set.add(banned[i]);
        }
        int count = 0;
        int sum = 0;
        for (int i = 1; i <= n; i++) {
            if (!set.contains(i)) {
                sum += i;
                count++;
                if (sum == maxSum) {
                    return count;
                }

                if (sum > maxSum) {
                    return count - 1;
                }
            }
        }
        return count;
    }
}

File: Problems/2554. Maximum Number of Integers to Choose From a Range I.py
"""
Problem Link: https://leetcode.com/problems/maximum-number-of-integers-to-choose-from-a-range-i/
"""


class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        banned = set(banned)
        count = 0
        for i in range(1, n + 1):
            if i not in banned:
                maxSum -= i
                if maxSum < 0:
                    break
                count += 1
        return count